Feat/int 36 toggle group (#333)
Atom-Learning/components: src/components/toggle-group/ToggleGroup.mdx

---
title: Toggle Group
component: ToggleGroup
description: Extends functionality the Toggle Group radix component
category: Content
---

Lets users toggle between a group of limited options(Suggested range is 2-6 options).

Functionality based on the [`ToggleGroup`](https://www.radix-ui.com/docs/primitives/components/toggle-group) radix component, which already allows for: single/multiple select, disabling or partly disabling options, adds keyboard navigation and orientation and more.

> Extends visually by allowing for vertical/horizontal display, gaps using stack or joined, rounded corners and so on.

Author: Aren't the examples below enough information regarding visual changes? 
I don't see the point of saying I'm adding a white background and primary colour font, I think that's unnecessary info to the user who can just see the examples etc.

> Extends visually by allowing for vertical/horizontal display, gaps using stack or joined, rounded corners and so on.

Reviewer: oh then the so on is unnecessary in my opinion. to make it more precise I would just switch to something like. I would just list the things the Dev can control with the variants
```suggestion
Extends the Radix component visually by allowing for vertical/horizontal display and gaps using stack or joined.
```

> Extends visually by allowing for vertical/horizontal display, gaps using stack or joined, rounded corners and so on.

Author: Okay - I'lll reword 👍

> Extends visually by allowing for vertical/horizontal display, gaps using stack or joined, rounded corners and so on.

Author: done - think it's cleaner now.

## Orientation

`orientation="vertical | horizontal"`

```jsx preview
<ToggleGroup.Root type="multiple" orientation="vertical">
  <ToggleGroup.Button value="a">A</ToggleGroup.Button>
  <ToggleGroup.Button value="b">B</ToggleGroup.Button>
</ToggleGroup.Root>
```

## Gap

`gap={0 | 1 | 2 | ... | 9}`

```jsx preview
<ToggleGroup.Root type="single" gap={3}>
  <ToggleGroup.Button value="a">A</ToggleGroup.Button>
  <ToggleGroup.Button value="b">B</ToggleGroup.Button>
</ToggleGroup.Root>
```

## Size

`size="sm | md | lg"`

```jsx preview
<ToggleGroup.Root type="single" gap={3} size="sm">
  <ToggleGroup.Button value="a">
    <Icon is={Upload} /> A
  </ToggleGroup.Button>
  <ToggleGroup.Button value="b">B</ToggleGroup.Button>
  <ToggleGroup.Button value="icon">
    <Icon is={Upload} />
  </ToggleGroup.Button>
</ToggleGroup.Root>
```

## Full width

> `isFullWidth={boolean}`

Reviewer: ```suggestion
`isFullWidth={true|false}`
```

just a small suggestion, as the other examples list the values not the type

Author: The other values expect an enum type of value though; I'm not necessarily against this - although boolean is always `true|false` - so I don't see a benefit to this change other than opinion

Reviewer: this is why I said it is a small suggestion.  it is just a consistency thing. not to precious on it though

Author: imma leave it as it is :)
